File: Php/core/Validation/NotNullValidator.php
namespace core\Validation{

	class NotNullValidator extends SingleValidator {

		protected $subject;
		protected $message;

		public function __construct($subject = null, $message){
			subject = $subject;
			message = $message;
		}

		public function validate(){
			if(subject == null){
				addError(message);
				return false;
			}
			return true;
		}

	}

}

File: Web/Validators/NotNullValidator.cs
namespace Web.Validators{

	class NotNullValidator : SingleValidator {
	    private readonly object _subject;
	    private readonly string _message;

	    public NotNullValidator(object subject, string message)
	    {
	        _subject = subject;
	        _message = message;
	    }

	    protected override void Validate(){
			if(_subject == null){
				AddError(_message);
			}
		}

	}

}